agent/pkg/proxy/module.go

```diff
@@ -0,0 +1,98 @@
+package proxy
+
+import (
+	"fmt"
+	"net"
+
+	"k8s.io/klog/v2"
+
+	"github.com/kubeedge/beehive/pkg/core"
+	"github.com/kubeedge/edgemesh/agent/pkg/common/informers"
+	"github.com/kubeedge/edgemesh/agent/pkg/common/modules"
+	"github.com/kubeedge/edgemesh/agent/pkg/common/util"
+	"github.com/kubeedge/edgemesh/agent/pkg/proxy/config"
+	"github.com/kubeedge/edgemesh/agent/pkg/proxy/controller"
+)
+
+// EdgeProxy is used for traffic proxy
+type EdgeProxy struct {
+	Config   *config.EdgeProxyConfig
+	APIConn  controller.ProxyController
+	Listener *net.TCPListener
+	Proxier  *Proxier
+}
+
+func newEdgeProxy(c *config.EdgeProxyConfig, ifm *informers.Manager) (proxy *EdgeProxy, err error) {
+	proxy = &EdgeProxy{Config: c}
+	if !proxy.Config.Enable {
+		return proxy, nil
+	}
```

Comment:
It should be coded like this:
```
if !c.Enable {
    return proxy, nil
}
proxy = &EdgeProxy{Config: c}
```
Otherwise, registryModules will terminate as an error.